File: takeMeFront/takeMeFront/src/types/VolunteerLeave.ts
/**
 * 请假信息 VolunteerLeave
 */

export interface VolunteerLeave {
  id: number                           // 主键ID
  volunteerId: number                  // 关联志愿者ID
  type: 0 | 1                          // 0=事假 1=病假
  startTime: string                     // 开始时间
  endTime: string                       // 结束时间
  reason: string                        // 请假原因
  status: 0 | 1 | 2                    // 0=待审核 1=已通过 2=已拒绝
  createTime: string                    // 提交时间
}

File: takeMe/pojo/src/main/java/com/me/entity/VolunteerLeave.java
package com.me.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("volunteer_leave")
public class VolunteerLeave {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long volunteerId;
    private Byte type;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String reason;
    private Byte status;
    private LocalDateTime createTime;
}